components/decimal/src/digit_char.rs

```diff
@@ -0,0 +1,14 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Algorithms to generate decimal digit characters.
+
+use std::convert::TryFrom;
+
+/// Returns the decimal digit with the given offset from the zero digit for a numbering system.
+pub fn get(zero_digit: char, value: u8) -> char {
+    // This should almost always succeed, but in case it fails (e.g. bad data), fall back
+    // gracefully to the Unicode replacement char U+FFFD.
+    char::try_from((zero_digit as u32) + (value as u32)).unwrap_or('\u{FFFD}')
```

Comment:
This works for all of the numeric numbering systems in CLDR:

https://github.com/unicode-cldr/cldr-core/blob/master/supplemental/numberingSystems.json

CLDR puts all other numbering systems in a bucket called "algorithmic".  In ICU, this is called RuleBasedNumberFormat (RBNF).  In the file, there is one numbering system, "hanidec", which works here: "〇一二三四五六七八九".  The other Chinese systems, like "hans", "hansfin", and "hanidays" require RBNF.

This is a large reason why I named this crate `icu_decimal` and the class `FixedDecimalFormat`: to emphasize that we're only dealing with decimal numbering systems, not algorithmic ones.

Here's a follow-up for RBNF: #595

Reply:
I will add this in a comment in this function, since it's important background information.

Reply:
Yeah I'm aware of RBNF -- hanidec was what I was thinking of; and those ideographs are not sequential in unicode

Reply:
Thanks for flagging this.  I will fix it!

Reply:
Okay, I fixed this by replacing `zero_digit: char` with `digits: [char; 10]`.  Good news: `cargo bench` reports a ~10% improvement in formatting speed.  The data size in JSON is a bit bigger, but I don't think this is a problem, because:

1. We are optimizing Bincode, not JSON, for data size, and Bincode should represent `[char; 10]` smaller than something like `Enum{char,[char;10]}`.
2. The code is simpler and faster with `[char; 10]`.
3. Since these files are small overall, we should be able to apply "full hunk deduplication" by having many locales point to the same data file.

Reply:
CC @gregtatum: FYI, I was wrong about zero_digit working for all Unicode numbering systems.  I saw that ICU had that optimization, and I didn't spend the time to check that it worked for everyone.